src/main/java/org/mockito/ArgumentCaptor.java

```diff
@@ -59,34 +61,17 @@
  * @since 1.8.0
  */
 public class ArgumentCaptor<T> {
-    
+
     HandyReturnValues handyReturnValues = new HandyReturnValues();
 
-    private final CapturingMatcher<T> capturingMatcher = new CapturingMatcher<T>();
+    private List<Object> arguments;
```

Comment:
That is (almost) impossible cause the type parameter `<T>` of `ArgumentMatcher `allows only to capture arguments for type `T`. The compile will prevent you from capturing a `Integer `with an `ArgumentCaptor<String>` unless you do some untyped casting.